docs/tools_multihost_hlo_runner.md

```diff
@@ -0,0 +1,147 @@
+# Multi-Host HLO Runner
+
+[TOC]
+
+This tool lets you run an HLO module on one or more GPUs. It also allows
+compiling code targeting multiple GPUs without running it.
+
+## Running multi-GPU (sharded) HLOs
+
+We can identify these HLOs by seeing `sharding=` annotations. For example
+`sharding={devices=[1,1,2,1]0,1}` means that the annotated tensor should be
+sharded to 2 GPUs (GPU0 and GPU1) along the 3rd dimension.
+
+The following instructions assume the working directory is the Tensorflow Git
+repository and that it had been ./configure'd.
+
+If we have enough GPUs, we can replay these HLOs like this:
+
+```
+bazel run -c opt --config=cuda --dynamic_mode=off \
+  //xla/tools/multihost_hlo_runner:hlo_runner_main -- my-hlo.txt
+```
+
+Tip: If the input generation takes too long or uses too much host memory,
+consider using `--hlo_argument_mode=uninitialized`.
+
+It is also possible to compile the same HLO without running it by setting
+`--run=false`
+
+```
+bazel run -c opt --config=cuda --dynamic_mode=off \
+  //xla/tools/multihost_hlo_runner:hlo_runner_main \
+  -- --run=false my-hlo.txt
+```
+
+In that case, a single GPU is necessary.
+
+
+### Troubleshooting
+- Errors such as `Check failed: result.replicas >= 1 (0 vs. 1)`:
+  -   We have to make sure that we have enough GPUs.
+  -   `CUDA_VISIBLE_DEVICES` must be set correctly or not set at all.
+-   Crashes:
+    -   We may want to use `--dynamic_mode=off`.
+    -   CUDA and Cudnn should be set up correctly.
+
+
+
+
+# Examples
+
+## Single process, multiple GPU example
+### Setup and get the HLO
+
+```
+docker run -it --shm-size=1g --gpus all ghcr.io/nvidia/jax:pax-2024-06-03
```

Comment:
I understand your point. But this is the example section. So I want them to just be able to copy/paste the results to run.

I'll refactor that a little bit with a warning. I'll put the "docker run; cd /opt/xla" in a "note" section that  warn them this can get outdated.
I'll keep that in a separate commit.

Reply:
I'd still think potentially outdated instructions are worse than no instructions at all, maybe move to a separate documentation page with "supported containers"?